runtime/src/accounts.rs

```diff
@@ -329,6 +329,33 @@ impl Accounts {
         versions.into_iter().map(|s| (s.0, s.2)).collect()
     }
 
+    pub fn load_by_program_slow(&self, ancestors: &HashMap<Fork, usize>, program_id: &Pubkey) -> Vec<(Pubkey, Account)> {
+        let mut accumulator: Vec<Vec<(Pubkey, u64, Account)>> = Vec::new();
+        for (fork, _) in ancestors.iter() {
+            let mut single_accumulator: Vec<Vec<(Pubkey, u64, Account)>> = self.accounts_db.scan_account_storage(
+                *fork,
+                |stored_account: &StoredAccount,
+                 _id: AppendVecId,
+                 accum: &mut Vec<(Pubkey, u64, Account)>| {
+                    if stored_account.balance.owner == *program_id {
+                        let val = (
+                            stored_account.meta.pubkey,
+                            stored_account.meta.write_version,
```

Comment:
this value is may be duplicated across forks, so you need another sort key (fork)

Reply:
actually, nevermind.  AppendVecId is apparently monotonically increasing on a per-accounts_db instance

Reply:
Ah, good point